src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs

```diff
@@ -0,0 +1,259 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+namespace Microsoft.Graph
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Reflection;
+    using System.Net.Http.Headers;
+
+    /// <summary>
+    /// Enum value to represent Microsoft graph service national cloud root endpoints
+    /// </summary>
+    public enum GraphServieCloudList {
+        /// <summary>
+        /// Global endpoint
+        /// </summary>
+        Global = 0,
+        /// <summary>
+        /// US_GOV endpoint
+        /// </summary>
+        US_GOV = 1,
+        /// <summary>
+        /// China endpoint
+        /// </summary>
+        China = 2,
+        /// <summary>
+        /// Germany endpoint
+        /// </summary>
+        Germany = 3
+    }
+
+    /// <summary>
+    /// GraphClientFactory class to create the HTTP client
+    /// </summary>
+    public static class GraphClientFactory
+    {
+        
+        /// The key for the SDK version header.
+        private static readonly string SdkVersionHeaderName = CoreConstants.Headers.SdkVersionHeaderName;
+
+        /// The version for current assembly.
+        private static Version assemblyVersion = typeof(GraphClientFactory).GetTypeInfo().Assembly.GetName().Version;
+
+        /// The value for the SDK version header.
+        private static string SdkVersionHeaderValue = string.Format(
+                    CoreConstants.Headers.SdkVersionHeaderValueFormatString,
+                    "Graph",
+                    assemblyVersion.Major,
+                    assemblyVersion.Minor,
+                    assemblyVersion.Build);
+
+        /// The default value for the overall request timeout.
+        private static readonly TimeSpan defaultTimeout = TimeSpan.FromSeconds(100);
+
+        /// The default value for the baseAddress of HTTP client
+        private static readonly Uri _baseAddress = new Uri("https://graph.microsoft.com/v1.0");
+
+        /// The default value for Api version
+        private static readonly string version = "v1.0";
+        
+        /// Microsoft Graph service nationa cloud endpoints
+        private static readonly string[] cloudList = { "https://graph.microsoft.com",
```

Comment:
Let's make this a Dictionary<TKey, TValue> so that we maintain association by key instead of order in the string array.

Reply:
Yes, it should be. Will update.

Reply:
If you change the enum into some static consts then we can use those as keys into the dictionary and it makes it extensible without bing a breaking change.